### PR TITLE
[release/6.0.3xx] introduced --use-program-main option for 6.0 console template to disable top-level program

### PR DESCRIPTION
**Pull request by vlada-shubina**
porting https://github.com/dotnet/templating/pull/4520 to 6.0.3xx for net6.0 project templates

### CODE REVIEW

File: template_feed/Microsoft.DotNet.Common.ProjectTemplates.6.0/content/ConsoleApplication-CSharp/.template.config/localize/templatestrings.cs.json
Comment (donJoseLuis) [reply template_feed/Microsoft.DotNet.Common.ProjectTemplates.6.0/content/ConsoleApplication-CSharp/.template.config/localize/templatestrings.cs.json:15]: looks like a pattern that we write the new content in English & someone else localizes it, is this correct?

**Review by baronfel — APPROVED** [review]

File: template_feed/Microsoft.DotNet.Common.ProjectTemplates.6.0/content/ConsoleApplication-FSharp/.template.config/ide.host.json
Comment (baronfel) [inline template_feed/Microsoft.DotNet.Common.ProjectTemplates.6.0/content/ConsoleApplication-FSharp/.template.config/ide.host.json:12]: Wait, the F# templates don't have/support UseProgramMain, should this be here?

**Comment by baronfel** [comment]
@vlada-shubina let's hold off on merge until we get approval from tactics, per @DamianEdwards's comment [here](https://github.com/dotnet/aspnetcore/pull/40945#issuecomment-1083344140)

**Review by phenning — APPROVED** [review]
Host file changes LGTM

**Pull request merged**
